fe/src/main/java/org/apache/doris/common/util/PropertyAnalyzer.java

```diff
@@ -371,4 +377,38 @@ public static long analyzeTimeout(Map<String, String> properties, long defaultTi
         }
         return timeout;
     }
+
+    public static Set<String> analyzeInvertedIndex(Map<String, String> properties,
+                                                   List<Column> columns) throws AnalysisException {
+        if (properties == null || !properties.containsKey(PROPERTIES_INVERTED_INDEX_COLUMNS)) {
+            return Sets.newHashSet();
+        }
+        String idxColsStr = properties.get(PROPERTIES_INVERTED_INDEX_COLUMNS);
+        properties.remove(PROPERTIES_INVERTED_INDEX_COLUMNS);
+        if (Strings.isNullOrEmpty(idxColsStr)) {
+            return Sets.newHashSet();
+        }
+        Set<String> resultCols = Sets.newHashSet();
+        Map<String, Column> columnMap = columns.stream()
+                .collect(Collectors.toMap(col -> col.getName().toLowerCase(), Function.identity()));
+
+        List<String> idxCols = COMMA_SPLITTER.splitToList(idxColsStr.toLowerCase());
+        for (String idxCol : idxCols) {
+            if (columnMap.containsKey(idxCol)) {
+                Column column = columnMap.get(idxCol);
+                if (column.isKey()) {
```

Comment:
here,
if the table is a agg table, inverted index can only be used in key columns, not value columns;
if the table is duplicate table, inverted index can be used in both key and value;
if the table is unique table, inverted index can only be used in key column.